### PR TITLE
Audio Effects Added & requirements.txt File

### PR DESCRIPTION
**Pull request by gacosta1**
Also started to add function for working with the optional two-factor authentication. Still working on that though

### CODE REVIEW

**Comment by gacosta1** [comment]
Update on PR

File: Software/scan_key/AuthenticationSQL.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 from on_off import *
 from ErrorSQL import *
 import os
+import pygame
```
Comment (ophilli) [inline Software/scan_key/AuthenticationSQL.py]: This is never used.

File: Software/scan_key/AuthenticationSQL.py [hunk 2]
```diff
@@ -49,23 +52,52 @@ def auth(id, pin, cursor):
 
     if(len(data)==0): #if there is no user with that data
```
Comment (ophilli) [inline Software/scan_key/AuthenticationSQL.py:52]: Where the hell is data coming from?

**Review by ophilli — COMMENTED** [review]
Can you clarify why these are two different functions? There is a LOT of code duplication between the two. To be honest - it looks to me like we should just have an authorization function with an if statement that runs the 2-3 LOC needed to check the PIN.

Will review more after response.

File: Software/scan_key/AuthenticationSQL.py (hunk 2 above)
Comment (gacosta1) [reply Software/scan_key/AuthenticationSQL.py:52]: lol copy & pasted from other func.

File: Software/scan_key/AuthenticationSQL.py (hunk 1 above)
Comment (gacosta1) [reply Software/scan_key/AuthenticationSQL.py]: was using a function from that library but instead used one from the os. So yeah we can delete that

Comment (gacosta1) [reply Software/scan_key/AuthenticationSQL.py]: To give a general answer to your question: So my idea was to have two functions. one that provides the keypad authentication if a  keypad is inserted and the other just for rfid.

File: Software/scan_key/AuthenticationSQL.py [hunk 3]
```diff
@@ -24,19 +24,21 @@ def machineAuth(id, cursor):
     for auth in authdata:
         if(auth[machineType] != 1):
             print ("USER IS NOT AUTHORIZED\n")
+            os.system("omxplayer deny.wav &")
```
Comment (ophilli) [inline Software/scan_key/AuthenticationSQL.py:27]: I'm well aware it is possible to be TOO abstract - but I wonder if it wouldn't be helpful down the road to have our own audio wrapper. Might make it easier to swap between .wav files or a Piezo buzzer.

File: Software/scan_key/AuthenticationSQL.py [hunk 4]
```diff
@@ -24,19 +24,21 @@ def machineAuth(id, cursor):
...
             errorSQL(id, 1)
             return (True)
 
     os.system("rm tmp")
```
Comment (ophilli) [inline Software/scan_key/AuthenticationSQL.py:31]: Are we anticipating generating a lot of tmp files? Is this just leftover code?

**Review by ophilli — COMMENTED** [review]
Where did deny.wav and successful.wav come from? What are the copyright licensing rules that let us use them?

File: Software/scan_key/AuthenticationSQL.py (hunk 3 above)
Comment (gacosta1) [reply Software/scan_key/AuthenticationSQL.py:27]: Okay I see where you're going with that. Have any current ideas on how we will implement the wrapper?

File: Software/scan_key/AuthenticationSQL.py (hunk 4 above)
Comment (gacosta1) [reply Software/scan_key/AuthenticationSQL.py:31]: the tmp file is a way for me to extract the name of the machine. I create a tmp file, extract the name from the file, and remove the file. It creates an extra process, but it does the job.

**Comment by ophilli** [comment]
os.system("hostname") should do the same thing I think.

**Comment by gacosta1** [comment]
deny.wav and successful.wav are free sound effects from the following links respectively.
https://www.freesound.org/people/suntemple/sounds/249300/
https://www.freesound.org/people/Kastenfrosch/sounds/162473/

Here is the copyright Info. from the sound effects
https://creativecommons.org/publicdomain/zero/1.0/

- You can find the information of licensed work on the two websites I posted above. It will be in a link saying "This work is licensed under the Creative Commons 0 License." On the right hand side of the site.

**Comment by gacosta1** [comment]
I'll try that.

**Comment by gacosta1** [comment]
@ophilli  I believe that os.system("hostname") will only output the hostname of the machine into stdio. I will not be able to assign it to as a value in the program. For example:
host = os.system("hostname")

**Comment by ophilli** [comment]
Perhaps the cleanest way is to call `name = subprocess.check_output(['hostname'])`

Subprocess is intended to replace OS.System

[More info.](https://docs.python.org/3.6/library/subprocess.html#module-subprocess)

**Comment by gacosta1** [comment]
Great! I will try that command when I can.

**Comment by gacosta1** [comment]
import subprocess

name = subprocess.check_output(['hostname'])
name = name.strip()
print name


Great it works! It does the job.

**Comment by ophilli** [comment]
Did a little more reading - sounds like sockets.gethostname() or platform.node() are actually the preferred methods for portability.

File: Software/scan_key/AuthenticationSQL.py (hunk 3 above)
Comment (ophilli) [reply Software/scan_key/AuthenticationSQL.py:27]: An aural feedback class would do the trick - just have a deny function that calls the appropriate soundtrack.

**Pull request closed without merging**